packages/nuxt/src/app/components/utils.ts

```diff
@@ -170,3 +171,37 @@ function isStartFragment (element: RendererNode) {
 function isEndFragment (element: RendererNode) {
   return element.nodeName === '#comment' && element.nodeValue === ']'
 }
+
+const componentWithTeleport = new WeakMap()
+
+/**
+ * Wrapper for components that should be teleported to islands placeholders
+ */
+export function withIslandTeleport (Component: DefineComponent) {
```

Comment:
```suggestion
export function withIslandTeleport (component: DefineComponent) {
```

Just a small nitpick, wouldn't `component` be a more appropriate variable name here?